src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -1,74 +1,119 @@
-    )
-    _alloc_full = ResPrototype(
-        "void* site_config_alloc_full(ext_joblist, env_varlist)", bind=False
-    )
-    _free = ResPrototype("void site_config_free( site_config )")
-    _get_installed_jobs = ResPrototype(
-        "ext_joblist_ref site_config_get_installed_jobs(site_config)"
-    )
-    _get_license_root_path = ResPrototype(
-        "char* site_config_get_license_root_path(site_config)"
-    )
-    _get_location = ResPrototype("char* site_config_get_location()", bind=False)
-    _get_config_file = ResPrototype("char* site_config_get_config_file(site_config)")
-    _get_env_var_list = ResPrototype(
-        "env_varlist_ref site_config_get_env_varlist(site_config)"
-    )
-
-    def __init__(self, config_content=None):
-
-        if config_content:
-            c_ptr = self._alloc(config_content)
-        else:
-            c_ptr = self._site_config_alloc_default()
-
-        if c_ptr is None:
-            raise ValueError("Failed to construct SiteConfig instance.")
-
-        super().__init__(c_ptr)
+from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
+
+
+@dataclass
+class SiteConfig:
+    job_list: ExtJoblist
+    env_vars: EnvironmentVarlist
 
     @classmethod
-    def from_config_dict(self, config_dict):
-        site_config = SiteConfig()
-        __license_root_path = site_config.get_license_root_path()
+    def _add_job(cls, job_list, license_root_path, job_path, job_name=None):
+        if not os.path.isfile(job_path):
+            print(f"WARNING: Unable to locate job file {job_path}")
+            return
+        new_job = ExtJob(
+            config_file=job_path,
+            private=False,
+            name=job_name,
+            license_root_path=license_root_path,
+        )
+        job_list.add_job(new_job.name(), new_job)
+
+    @classmethod
+    def _add_config_content(
+        cls,
+        license_root_path: str,
+        config_content: ConfigContent,
+        job_list: ExtJoblist,
+        env_vars: EnvironmentVarlist,
+    ):
+
+        if config_content.hasKey(ConfigKeys.INSTALL_JOB):
+            for args in iter(config_content[ConfigKeys.INSTALL_JOB]):
+                cls._add_job(job_list, license_root_path, args[1], args[0])
+
+        if config_content.hasKey(ConfigKeys.INSTALL_JOB_DIRECTORY):
+            for args in iter(config_content[ConfigKeys.INSTALL_JOB_DIRECTORY]):
+                job_path = args[0]
+                if not os.path.isdir(job_path):
+                    print(f"WARNING: Unable to locate job directory {job_path}")
+                    continue
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    cls._add_job(job_list, license_root_path, full_path)
+
+        environment_vars = config_content[ConfigKeys.SETENV]
+
+        for elem in iter(environment_vars):
+            env_vars.setenv(elem[0], elem[1])
```

Comment:
Is this possible?
``` python
for key, value in iter(environment_vars):
    env_vars.setenv(key, value)
```
in place of 
``` python
for elem in iter(environment_vars):
    env_vars.setenv(elem[0], elem[1])
```
If that is ok and you also like it you can apply the same thing for ln 58-59

Reply:
I think that is a ContentNode, and probably does not support that sort of thing

Reply:
I think it will support this, and it should be safe. The parser should enforce that the elements of `environment_vars` have exactly two elements in them. If we keep the `elem[0]`, `elem[1]` we will have a silent failure if there is a `elem[2]`. So I think Dan's suggestion is preferable here.

Reply:
But we should probably test that we get a reasonable error message. I will add a test for that.

Reply:
@eivindjahren says it will work